src/tests/Microsoft.PowerFx.Core.Tests.Shared/TexlTests.cs

```diff
@@ -1,4457 +1,4487 @@
+        [InlineData("DateAdd([Date(2000,1,1)],[3])", "*[Value:D]")]
+        [InlineData("DateAdd(Table({a:Date(2000,1,1)}),[3])", "*[a:D]")]
+        [InlineData("DateAdd(Date(2000,1,1),[1])", "*[Result:D]")]
+        [InlineData("DateAdd(\"2021-02-03\",[1])", "*[Result:d]")] // Coercion from string
+        [InlineData("DateAdd(44955,[1])", "*[Result:d]")] // Coercion from number
+        [InlineData("DateAdd(Time(12,0,0),[1])", "*[Result:T]")] // Coercion from time
+        [InlineData("DateAdd([DateTimeValue(\"1 Jan 2015\")],1)", "*[Value:d]")]
+        [InlineData("DateAdd([DateTimeValue(\"1 Jan 2015\")],[3])", "*[Value:d]")]
+        [InlineData("DateAdd(DateTimeValue(\"1 Jan 2015\"),[1])", "*[Result:d]")]
+        [InlineData("DateAdd([\"2011-02-03\"],1)", "*[Value:d]")] // Coercion from string
+        [InlineData("DateAdd([44900],1)", "*[Value:d]")] // Coercion from number
+        [InlineData("DateAdd([Time(12,0,0)],1)", "*[Value:T]")] // Coercion from time
+        [InlineData("DateDiff([Date(2000,1,1)],[Date(2001,1,1)],\"years\")", "*[Result:n]")]
+        [InlineData("DateDiff(Date(2000,1,1),[Date(2001,1,1)],\"years\")", "*[Result:n]")]
+        [InlineData("DateDiff([Date(2000,1,1)],Date(2001,1,1),\"years\")", "*[Result:n]")]
+        public void TexlDateTableFunctions_Float(string expression, string expectedType)
+        {
+            var engine = new Engine(new PowerFxConfig(Features.None));
+            var options = new ParserOptions() { NumberIsFloat = true };
+            var result = engine.Check(expression, options);
+
+            Assert.True(DType.TryParse(expectedType, out var expectedDType));
+            Assert.Equal(expectedDType, result.Binding.ResultType);
+            Assert.True(result.IsSuccess);
+        }
+
+        [Theory]
+        
+        // Collect
+        [InlineData("Collect(t1, {a:1})", "![a:w, b:s]")]
+        [InlineData("Collect(t1, {a:1},{a:2})", "*[a:w, b:s]")]
+        [InlineData("Collect(t1, Table({a:1}))", "*[a:w, b:s]")]
+        [InlineData("Collect(t2, 2)", "![Value:w]")]
+        [InlineData("Collect(t2, 2, 3, 4)", "*[Value:w]")]
+        [InlineData("Collect(t1, {a:1})", "*[a:w, b:s]", false)]
+        [InlineData("Collect(t1, {a:1},{a:2})", "*[a:w, b:s]", false)]
+        [InlineData("Collect(t1, Table({a:1}))", "*[a:w, b:s]", false)]
+        [InlineData("Collect(t2, 2)", "*[Value:w]", false)]
+        [InlineData("Collect(t2, 2, 3, 4)", "*[Value:w]", false)]
+        public void TexlMutationFunctionsV1Tests(string expression, string expectedType, bool isPFxV1 = true)
+        {
+            var engine = new Engine(new PowerFxConfig(isPFxV1 ? Features.PowerFxV1 : Features.None));
+            var options = new ParserOptions() { AllowsSideEffects = true };
+
+            DType.TryParse("*[a:w,b:s]", out var expectedDType);
+            DType.TryParse("*[Value:w]", out var expectedDTypeScalar);
+
+            engine.Config.SymbolTable.AddFunction(new CollectFunction());

```

Comment:
[minor] can we just call EnableMutationFunctions instead of adding each function individually? If we don't call those, then we don't need the Patch functions either.

Reply:
`EnableMutationFunctions` is an extension method from the Interpreter. These are Core tests.